### PR TITLE
Implement emotion & chakra

### LINKED ISSUE #7054: Migrate from styled-components to emotion to prepare the ground for Chakra

### PR DESCRIPTION
**Pull request by pettinarip**
## Description

The first step towards adopting Chakra UI is to migrate from `styled-components` to `emotion`.

**UPDATE:** chakra PR was merged #7195 **now** this PR includes chakra + emotion setup.

## Related Issue

Epic #6374
Fixes #7054

### CODE REVIEW

File: docs/best-practices.md [hunk 1]
```diff
@@ -111,7 +111,7 @@ export default ComponentName
   ```tsx
   // Example of styling syntax using styled-components
```
Comment (samajammin) [inline docs/best-practices.md:112]:
```suggestion
  // Example of styling syntax using emotion
```
There's also a mention of styled-components (& link to https://styled-components.com/) above on line 107 - let's update this to reflect that we're now using emotion.

File: docs/best-practices.md [hunk 2]
```diff
@@ -111,7 +111,7 @@ export default ComponentName
...
 
-  import styled from "styled-components"
+  import styled from "@emotion/styled"
 
   const GenericButton = styled.div`
     width: 200px;
```
Comment (samajammin) [inline docs/best-practices.md:117]: Another mention below this (line 131) that should be updated:

> Recommended VS Code Plugin: `vscode-styled-components`

**Comment by samajammin** [comment]
Looks great overall @pettinarip! Noticed a few small styling issues:

1) header anchor links (both desktop & mobile)

Image 2022-08-01 at 2 20 51 PM

2) overall spacing of pages & size of headers

Spacing of content on docs pages looks off:
Image 2022-08-01 at 2 23 48 PM

Same with tutorials:
Image 2022-08-01 at 2 24 33 PM

Same with static templates:
Image 2022-08-01 at 2 24 51 PM

Most product pages look great but there's some slight differences in spacing / header font size on some pages:

Image 2022-08-01 at 2 25 52 PM

**Review by samajammin — CHANGES_REQUESTED** [review]
Great work! Some minor styling issues & some documentation updates needed.

**Comment by minimalsm** [comment]
Styling changes here look to be fixed in this PR: #7195

File: docs/best-practices.md (hunk 1 above)
Comment (pettinarip) [reply docs/best-practices.md:112]: Cool, will create a task to rewrite all this documentation once we merge the Chakra PR. Should be kind of aligned with what we are writing in the migration guide.

File: src/@chakra-ui/gatsby-plugin/styles.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+const styles = {
+  global: (_props) => ({
+    /**
+     * Current scenario: we have 2 palces where global styles are defined.
```
Comment (wackerow) [inline src/@chakra-ui/gatsby-plugin/styles.ts:4]:
```suggestion
     * Current scenario: we have 2 places where global styles are defined.
```

File: src/components/FileContributors.tsx [hunk 4]
```diff
@@ -12,10 +13,14 @@ import { ButtonSecondary } from "./SharedStyledComponents"
 import { getLocaleTimestamp } from "../utils/time"
 import { Lang } from "../utils/languages"
 
-const loadingStyles = css`
+const loadingStyles = (theme: Theme) => css`
```
Comment (wackerow) [inline src/components/FileContributors.tsx:16]: 👍

File: src/components/Layout.tsx [hunk 5]
```diff
@@ -96,15 +99,6 @@ const Layout: React.FC<IProps> = ({
...
 
-  // set isDarkTheme based on browser/app user preferences
-  useEffect(() => {
-    if (localStorage && localStorage.getItem("dark-theme") !== null) {
-      setIsDarkTheme(localStorage.getItem("dark-theme") === "true")
-    } else {
-      setIsDarkTheme(window.matchMedia("(prefers-color-scheme: dark)").matches)
-    }
-  }, [])
-
```
Comment (wackerow) [inline src/components/Layout.tsx:107]: @pettinarip So am I correct the users browser preference will still be accounted for here?

**Review by wackerow — APPROVED** [review]
🎉 Nice job @pettinarip!

File: src/components/Layout.tsx (hunk 5 above)
Comment (pettinarip) [reply src/components/Layout.tsx:107]: Yes, we are using this flag `useSystemColorMode` for that. More on this here: https://chakra-ui.com/docs/styled-system/color-mode#updating-the-theme-config

**Comment by samajammin** [comment]
Exciting! 😄

**Pull request merged**